frame/merkle-mountain-range/src/mmr/utils.rs

```diff
@@ -0,0 +1,132 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Merkle Mountain Range utilities.
+
+/// MMR nodes & size -related utilities.
+pub struct NodesUtils {
+	no_of_leaves: u64,
+}
+
+impl NodesUtils {
+	/// Create new instance of MMR nodes utilities for given number of leaves.
+	pub fn new(no_of_leaves: u64) -> Self {
+		Self { no_of_leaves }
+	}
+
+	/// Calculate number of peaks in the MMR.
+	pub fn number_of_peaks(&self) -> u64 {
+		self.number_of_leaves().count_ones() as u64
+	}
+
+	/// Return the number of leaves in the MMR.
+	pub fn number_of_leaves(&self) -> u64 {
+		self.no_of_leaves
+	}
+
+	/// Calculate the total size of MMR (number of nodes).
+	pub fn size(&self) -> u64 {
+		2 * self.no_of_leaves - self.number_of_peaks()
+	}
+
+	/// Calculate maximal depth of the MMR.
+	pub fn depth(&self) -> u32 {
+		if self.no_of_leaves == 0 {
+			return 0
+		}
+
+		64 - self.no_of_leaves
+				.next_power_of_two()
+				.to_le()
```

Comment:
I don't understand why this `to_le` needed. Can you explain?

Reply:
@pepyakin cleared that up for me in private conversation, it was indeed unnecessary and even harmful on BE platforms (since it was equivalent of calling `.swap_bytes()`).